tests/pytorch/test_numerics.py

```diff
@@ -2976,3 +2976,75 @@ def _run_module(m, inp):
     out = _run_module(g2, b)
 
     assert_allclose(out, outT, 1e-7)
+
+
+@pytest.mark.parametrize(
+    "quantization_type",
+    [
+        "fp8",
+        "mxfp8",
+    ],
+)
+@pytest.mark.parametrize(
+    "shape,chunks,dim",
+    [
+        ((64, 128), 2, 0),  # Split along first dimension, needs padding for mxfp8
+        ((64, 128), 2, 1),  # Split along second dimension, goes down dequantization path for mxfp8
+    ],
+)
+def test_fp8_split_functionality(quantization_type, shape, chunks, dim):
+    """Test torch.chunk on FP8 and MXFP8 tensors and verify correctness via dequantization."""
+    if quantization_type == "fp8" and not fp8_available:
+        pytest.skip(reason_for_no_fp8)
+    if quantization_type == "mxfp8" and not mxfp8_available:
+        pytest.skip(reason_for_no_mxfp8)
+
+    device = "cuda"
+    dtype = torch.bfloat16
+
+    # Create reference tensor
+    torch.manual_seed(1234)
+    torch.cuda.manual_seed(1234)
+    ref_tensor = torch.randn(shape, device=device, dtype=dtype)
+
+    # Quantize the tensor
+    if quantization_type == "fp8":
+        quantizer = Float8Quantizer(
+            scale=torch.ones(1, dtype=torch.float32, device=device).squeeze(),
+            amax=torch.zeros(1, dtype=torch.float32, device=device),
+            fp8_dtype=tex.DType.kFloat8E4M3,
+        )
+        quantized_tensor = quantizer(ref_tensor)
+    elif quantization_type == "mxfp8":
+        quantizer = MXFP8Quantizer(fp8_dtype=tex.DType.kFloat8E4M3)
+        quantized_tensor = quantizer(ref_tensor)
+
+    # Apply torch.chunk on quantized tensor
+    quantized_tensor_dispatch_out = torch.chunk(quantized_tensor, chunks, dim=dim)
+    # need to make tensor contigous for dim=1 splitting.
+    outs = [out.contiguous() for out in quantized_tensor_dispatch_out]
```

Comment:
Can we move this `contiguous` into `dequantize`? Normal `torch.Tensor`s work without it, so it's a problem in our implementation if we force users do this extra unintuitive step.